src/main/java/core/basesyntax/dao/WarehouseImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.dao;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class WarehouseImpl implements Warehouse {
+    private Map<String, Integer> listFruits = new HashMap<>();
```

Comment:
In my opinion, it would be better to implement class Fruit with fields String name and int amount. Then you should create a list of fruits but not map. Let's imagine that you want to add to your shop vegetables. You need to create one more map, so your program is opened for modifications.